src/Grid/grid.jl

```diff
@@ -68,12 +68,133 @@ struct VertexIndex <: BoundaryIndex
     idx::Tuple{Int,Int} # cell and side
 end
 
+struct Neighbor{T<:BoundaryIndex}
+    neighbor_info::Vector{T}
+end
```

Comment:
I think we should rename this one to something like `Neighborhood`, `GeneralNeighborhood` or something, because using a singular name is a bit misleading. No idea how to improve the data member's name.

Further I would introduce some specialized types, i.e. FaceNeighborhood, EdgeNeighborhood and VertexNeighborhood.

Since I just noted, is there some rationale behind wrapping this one in a struct? It looks a bit like direclty using the Vector could be sufficient.

Reply:
> Since I just noted, is there some rationale behind wrapping this one in a struct? It looks a bit like direclty using the Vector could be sufficient.


The reason behind wrapping it in a struct is that I want to dispatch `Base.:+` see https://github.com/Ferrite-FEM/Ferrite.jl/pull/363/files#diff-13ae190cb008d749905405bddca2a9ce395b303e12c781186df0d908efd05a90R83-R86

We need this because if we construct `I, J, V` of each sparse matrix, `+` is called if more than one `V`entry exists at the same `I` `J` indices. Probably a `Base.:+(x::Vector{T}, y::Vector{T}) where T<:BoundaryIndex` type piracy is possible but I thought we should avoid it, if possible

Reply:
Regarding naming, we should agree one something here before I change it in the code: maybe `EntityNeighborhood` ?